### ISSUE TITLE
Monitor other processes/commands

### ISSUE THREAD

**Issue by barryvdh** [laravel/horizon#342]
Would it be possible to let Horizon also monitor (and possibly restart) other commandos? Thinking about socket chat server, or a long running artisan command that is listening to a socket etc? Especially when it's related to Laravel it would be nice to be able to keep running and also restart, when code changes.

**Comment by driesvints** [comment]
I believe this goes a little beyond what Horizon is intended to do which is managing your queues. I'll flag this as a review for Taylor to see what he thinks.

**Comment by halaei** [comment]
IMO, this should be a feature of the framework itself. The exact feature I have is to monitor any closure, to make sure it is always calling and gets called again after termination, until some command to restart is issued, like queue:restart or horizon:terminate. Of course, horizon adds to it process monitoring as well.
The reason I think this should be in framework is that every time I needed something like that, I went to copy the code in laravel daemon queue, and generalize/customize it for my specific application. I think this is what happened in horizon as well, parts of the code is copy of daemon queue, in some degree.
I remembered I sent a PR to laravel a while ago (before horizon) to generalize daemon. But the pull request get rejected. Then I added it into my own helper package. I don't think I ever used it though I really need to try it out.
Here is the pull request: https://github.com/laravel/framework/pull/18614
Here is my implementation: https://github.com/halaei/helpers/tree/master/src/Supervisor. Check the docs if you are interested: https://github.com/halaei/helpers#supervisor
